api/models/cartDao.js

```diff
@@ -0,0 +1,105 @@
+const dataSource = require('./data-source')
+
+const createCartList = async (userId, itemId, quantity, optionId) => {
+    const result = await dataSource.query(
+        `INSERT INTO carts(
+            user_id,
+            item_id,
+            quantity,
+            option_id
+        ) VALUES (?, ?, ?, ?)
+        `, [userId, itemId, quantity, optionId]
+    )
+
+    return result;
+}
+
+const getAllCartList = async (userId, limit, offset) => {
+    const result = await dataSource.query(
+        `SELECT
+            i.id,
+            i.name,
+            i.price,
+            i.detail_image,
+            c.quantity,
+            c.checkbox,
+            o.name as option_name
+        FROM carts as c
+        LEFT JOIN items as i ON i.id = c.item_id
+        LEFT JOIN options as o ON o.id = c.option_id
+        WHERE c.user_id = ?
+        LIMIT ? OFFSET ?
+        `, [userId, limit, offset]
+    )
+
+    return result;
+}
+
+const checkCart = async (userId, itemId, optionId) => {
+    const [result] = await dataSource.query(
+        `SELECT EXISTS  (
+            SELECT id 
+            FROM carts
+            WHERE user_id = ?
+            AND item_id = ?
+            AND (
+                IF (
+                    ? = 1,
+                    option_id = 1,
+                    option_id IS NULL
+                )
+            )
+        ) AS is_existed
```

Comment:
select에서 사용한 IF가 아니므로 뒤에 AS가 없어도 될 것 같은데 한 번 확인해 주세요.
```suggestion
        ) 
```

Reply:
삭제하니 리턴되는 결과에 쿼리문이 찍혀서 일단은 두도록 하겠습니다!